### DISCUSSION TITLE
Is there anyway to realize strong-coupling calculation between two different physical domains

### DISCUSSION THREAD

**Discussion by ruihuasun** [mfem/mfem#3399] [Q&A]
I want to, for example, calculate the strong-coupling of structural vibration with the sound field in the air. Strong-coupling means that we solve PDEs all at once after substituting coupling equations, rather than a weak-coupling: solve structural PDEs, than sound field PDEs, than structural,...

I noticed the submesh in v4.5 recently but it seems that it can only be used for weak-coupling problem. Maybe I didn't fully understand the submesh. I also noticed #2136 , but still no feasible solution. Besides, no such example can be found in examples list.

Is there anyone could help me?

**Comment by jandrej** [comment]
Strong coupling can be achieved by formulating an algebraic block system from subdomains and introducing essential boundary conditions. There is no example for this and it is a little more involved than weak-coupling but not terribly complicated.

**Reply by ruihuasun** [reply]
Really thanks for ur reply. This is still too difficult for me, I really hope to have an example reference. (Consider that I have studied mfem for a while, maybe it is 'terribly' harder for new participants.)
